### DISCUSSION TITLE
(POLL) We're developers... let's make our own GitHub Feed (+ how to bring the old one back)

### DISCUSSION THREAD

**Discussion by SamJakob** [community/community#66249] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

If GitHub isn't going to give us the homepage feed we want, why don't we create our own!

(If you happen upon this, feel free to leave a comment saying what you would like to see on the feed)

### To GitHubbers

(If anyone at GitHub sees this, **please** restore the old feed or let us **_optionally_** have a more powerful 'dashboard' that we can customize to _our liking_ _**in addition to the old feed**_ as opposed to whatever this is now...)

The old feed worked for so many without issue - myself included - and was something of comfort for most developers who use GitHub for their job or in their spare time and allowed them to access **relevant** information about what's going on. People have grown accustomed to that specific layout and plan their workflow or routine around it.

My two cents is that changes to the homepage feed of this site should be approached as a power tool for interested users, not a new default that changes things to be less intuitive and less relevant.

**Comment by leeoniya** [comment]
i looked at octokit (the official GH api sdk)

i dont think it's feasible to query all 300 devs i follow and 1k repos i watch for all activity, then filter myself. im sure it will be some garbage rate limited thing, even with GraphQL queries. hope im wrong 🤷

**Reply by SamJakob** [reply]
I'm not sure what the rate limits actually are for REST or GraphQL - or the limits on the depth of a GraphQL query - but I imagine on the face of it that especially with some fairly trivial caching it might be quite feasible? It seems like you can request up to 100 items per page for paginated requests.

EDIT: managed to find some details: https://docs.github.com/en/graphql/overview/resource-limitations#rate-limit
I think some experimentation would be needed to figure out the economy of making the calls. It might be possible to cut them down and optimize them and some other techniques could get involved.

This page lists some approaches I was entertaining for solving the issue (https://www.endorlabs.com/blog/how-to-get-the-most-out-of-github-api-rate-limits).

Also if a back-end service was introduced in an acceptable way, it could slowly pull the data in over time or during inactive hours.

**Reply by port19x** [reply]
That's where caching comes in handy :wink:

**Comment by SamJakob** [comment]
I've jumped on the bandwagon for now of coming up with ways to restore the original.

## My browser extension to restore the old feed

- **macOS and iOS Safari now supported** (see the repo link below)
- **Firefox version now available on Firefox Extension Marketplace**
- **Chrome version now available on Chrome Web Store**
  (see the repo README linked below)

https://github.com/SamJakob/github-homepage

(I preferred a standalone browser extension as anyone should be able to inspect the full source of it and be confident in its safety, as opposed to analyzing a whole userscripts extension - but that's just personal preference!)

## Alternatives (Userscripts)
These should be cross-browser if you can find a userscripts extension for your browser.

There's also this userscript here:
https://github.com/orgs/community/discussions/66260

...and someone else's here:
https://github.com/Gerrit0/old-github-feed

...and this one too:
https://github.com/wangrongding/github-old-feed
